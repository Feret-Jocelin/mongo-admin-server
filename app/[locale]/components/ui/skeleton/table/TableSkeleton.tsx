import styles from './tableSkeleton.module.scss';

const TableSkeleton = () => {
  const colnm = 4;
  const rownm = 6;

  return (
    <table className={styles['table-skeleton']}>
      <thead>
        <tr>
          {Array.from(Array(colnm).keys()).map((_, index) => (
            <th key={index}>
              <div></div>
            </th>
          ))}
        </tr>
      </thead>
      <tbody className={styles['table-skeleton--tbody']}>
        {Array.from(Array(rownm).keys()).map((_, i) => (
          <tr key={i} className={styles['table-skeleton--tbody--tr']}>
            {Array.from(Array(colnm).keys()).map((_, j) => (
              <td key={j} className={styles['table-skeleton--tbody--tr--td']}>
                <div
                  className={styles['table-skeleton--tbody--tr--td--div']}
                ></div>
              </td>
            ))}
          </tr>
        ))}
      </tbody>
    </table>
  );
};

export default TableSkeleton;
