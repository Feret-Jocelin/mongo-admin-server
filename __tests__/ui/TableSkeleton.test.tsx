/* eslint-disable no-undef */
import React from 'react';
import { render, screen } from '@testing-library/react';
import TableSkeleton from '@/app/[locale]/components/ui/skeleton/table/TableSkeleton';

describe('TableSkeleton', () => {
  it('renders', () => {
    const { container } = render(<TableSkeleton />)
    const table = screen.getByRole('table');
    expect(table).toBeInTheDocument();
    expect(table).toHaveClass('table-skeleton')
    expect(container).toMatchSnapshot();
  })
})