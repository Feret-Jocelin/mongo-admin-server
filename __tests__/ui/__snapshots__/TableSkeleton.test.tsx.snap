// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableSkeleton renders 1`] = `
<div>
  <table
    class="table-skeleton"
  >
    <thead>
      <tr>
        <th>
          <div />
        </th>
        <th>
          <div />
        </th>
        <th>
          <div />
        </th>
        <th>
          <div />
        </th>
      </tr>
    </thead>
    <tbody
      class="table-skeleton--tbody"
    >
      <tr
        class="table-skeleton--tbody--tr"
      >
        <td
          class="table-skeleton--tbody--tr--td"
        >
          <div
            class="table-skeleton--tbody--tr--td--div"
          />
        </td>
        <td
          class="table-skeleton--tbody--tr--td"
        >
          <div
            class="table-skeleton--tbody--tr--td--div"
          />
        </td>
        <td
          class="table-skeleton--tbody--tr--td"
        >
          <div
            class="table-skeleton--tbody--tr--td--div"
          />
        </td>
        <td
          class="table-skeleton--tbody--tr--td"
        >
          <div
            class="table-skeleton--tbody--tr--td--div"
          />
        </td>
      </tr>
      <tr
        class="table-skeleton--tbody--tr"
      >
        <td
          class="table-skeleton--tbody--tr--td"
        >
          <div
            class="table-skeleton--tbody--tr--td--div"
          />
        </td>
        <td
          class="table-skeleton--tbody--tr--td"
        >
          <div
            class="table-skeleton--tbody--tr--td--div"
          />
        </td>
        <td
          class="table-skeleton--tbody--tr--td"
        >
          <div
            class="table-skeleton--tbody--tr--td--div"
          />
        </td>
        <td
          class="table-skeleton--tbody--tr--td"
        >
          <div
            class="table-skeleton--tbody--tr--td--div"
          />
        </td>
      </tr>
      <tr
        class="table-skeleton--tbody--tr"
      >
        <td
          class="table-skeleton--tbody--tr--td"
        >
          <div
            class="table-skeleton--tbody--tr--td--div"
          />
        </td>
        <td
          class="table-skeleton--tbody--tr--td"
        >
          <div
            class="table-skeleton--tbody--tr--td--div"
          />
        </td>
        <td
          class="table-skeleton--tbody--tr--td"
        >
          <div
            class="table-skeleton--tbody--tr--td--div"
          />
        </td>
        <td
          class="table-skeleton--tbody--tr--td"
        >
          <div
            class="table-skeleton--tbody--tr--td--div"
          />
        </td>
      </tr>
      <tr
        class="table-skeleton--tbody--tr"
      >
        <td
          class="table-skeleton--tbody--tr--td"
        >
          <div
            class="table-skeleton--tbody--tr--td--div"
          />
        </td>
        <td
          class="table-skeleton--tbody--tr--td"
        >
          <div
            class="table-skeleton--tbody--tr--td--div"
          />
        </td>
        <td
          class="table-skeleton--tbody--tr--td"
        >
          <div
            class="table-skeleton--tbody--tr--td--div"
          />
        </td>
        <td
          class="table-skeleton--tbody--tr--td"
        >
          <div
            class="table-skeleton--tbody--tr--td--div"
          />
        </td>
      </tr>
      <tr
        class="table-skeleton--tbody--tr"
      >
        <td
          class="table-skeleton--tbody--tr--td"
        >
          <div
            class="table-skeleton--tbody--tr--td--div"
          />
        </td>
        <td
          class="table-skeleton--tbody--tr--td"
        >
          <div
            class="table-skeleton--tbody--tr--td--div"
          />
        </td>
        <td
          class="table-skeleton--tbody--tr--td"
        >
          <div
            class="table-skeleton--tbody--tr--td--div"
          />
        </td>
        <td
          class="table-skeleton--tbody--tr--td"
        >
          <div
            class="table-skeleton--tbody--tr--td--div"
          />
        </td>
      </tr>
      <tr
        class="table-skeleton--tbody--tr"
      >
        <td
          class="table-skeleton--tbody--tr--td"
        >
          <div
            class="table-skeleton--tbody--tr--td--div"
          />
        </td>
        <td
          class="table-skeleton--tbody--tr--td"
        >
          <div
            class="table-skeleton--tbody--tr--td--div"
          />
        </td>
        <td
          class="table-skeleton--tbody--tr--td"
        >
          <div
            class="table-skeleton--tbody--tr--td--div"
          />
        </td>
        <td
          class="table-skeleton--tbody--tr--td"
        >
          <div
            class="table-skeleton--tbody--tr--td--div"
          />
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
